### ISSUE TITLE
importing a function from a child-component breaks on:click binding to the child-component element

### ISSUE THREAD

**Issue by JaneOri** [canjs/can-stache-bindings#515]
Broken example: https://codepen.io/anon/pen/ZZzNyV?editors=1010

broken snippet:
```js
// child-component is defined and their methods can be imported to the parent
Component.extend({
  tag: "my-counter",
  view: `
    Parent Count: <span>{{this.count}}</span><br>
    <button on:click="this.importedfn()">call importedfn from child to increase child count</button>
    <br>
    <child-component exportme:to="importedfn" on:click="increment()" />
  `,
  ViewModel: {
    count: {default: 0},
    importedfn: {default: undefined},
    increment() {
      this.count++;
    }
  }
});
```
the `on:click` attached to child-component's element in the parent (`my-counter`) view does not work.

Removing `exportme:to="importedfn"` from child-component's element in the parent view allows the on:click method to function as expected.

No error is thrown, I suspect it's listening for a "click" event from the component rather than the dom. ~~As far as I could find, there is no way to specify that the binding should specifically be to the dom events like `on:event:by:dom` which would at least help me get unstuck. (any temporary workarounds appreciated)~~

edit:
`on:el:click="increment()"` is a workaround (that's [not documented](https://canjs.com/doc/can-stache-bindings.event.html), I discovered it in the tests) and suggests that the default binding behavior for on:click is changed from domEvent to component event when a function is exported

**Comment by justinbmeyer** [comment]
@James0x57 just quickly glancing at this ... is the only problem that `<custom-element on:click="method()"`> binds to the `ViewModel` instead of the element? 

This is expected behavior (at least expected by CanJS, maybe not by users).  All bindings on a component, data and event, are done on the ViewModel by default.

**Comment by justinbmeyer** [comment]
Regarding the documentation ... it was documented, but I removed it as it was uncommonly used. We might need to bring it back, but tucked away in a corner ...

**Comment by JaneOri** [comment]
@justinbmeyer `on:click` only binds to the the ViewModel when a function is also exported.
It binds to the dom event (as I expected at the time it should) when the function is not exported.

(by exported I mean `<child-component imExported:to="thatExportedFunction">`)


------
tl;dr:

on:click binds to DOM event (expected):
 `<child-component on:click="doSomething()">`

on:click binds to ViewModel:
 `<child-component on:click="doSomething()" imExported:to="thatExportedFunction">`

on:click binds to DOM event (but option to do this is undocumented):
 `<child-component on:el:click="doSomething()" imExported:to="thatExportedFunction">`

**Comment by bmomberger-bitovi** [comment]
This was documented in CanJS 4.  https://v4.canjs.com/doc/can-stache-bindings.event.html#on_el_DOM_EVENT__CALL_EXPRESSION_

In the current (5.x) docs those patterns are gone.  A line about :el: and :vm: was also removed from the top level can-stache-bindings doc page.

**Comment by bmomberger-bitovi** [comment]
Because the :el: and :vm: cases still exist in the doc pages for :from, :to, and :bind, I'm guessing that this is an oversight in reorganizing the docs and we should still have cases for on:.  Can @justinbmeyer confirm?

**Comment by bmomberger-bitovi** [comment]
I am of the opinion that we should bring back the documentation for :el: and :vm: for events.  Listening on an element is useful to users because bubbling the event up the DOM avoids a lot of tedious linking of event handlers in intervening components (VM events don't bubble automatically).

I'm not saying we should recommend this to users, but _as a user_, I have found this pattern necessary.

**Comment by bmomberger-bitovi** [comment]
> Removing `exportme:to="importedfn"` from child-component's element in the parent view allows the on:click method to function as expected.

I'm not seeing what is described:  When `exportme:to="importedfn"` is removed, **neither** counter updates.

**Comment by bmomberger-bitovi** [comment]
In both cases, `document.querySelector("child-component").viewModel.dispatch("click")` will update the parent count.  I don't think there's a bug here, inasmuch as it matches the documented-in-CanJS-4 behavior. Also I tried as far back as CanJS 5.19 and couldn't replicate James's result.

EDIT: I only tried to replicate the result on Chrome so far.  May try other browsers today.

**Comment by bmomberger-bitovi** [comment]
Notes from office hours.
- it would be a good idea to warn if the event with a well known name (e.g. "click") is bound to a VM.
- for Can 6 and beyond, might want to decide whether to use el or vm by default differently if the event is a known DOM event.

**Comment by chasenlehara** [comment]
I think this was addressed in [v4.10.5](https://github.com/canjs/can-stache-bindings/releases/tag/v4.10.5).
